lib/npg_qc/Schema/ResultSet.pm

```diff
@@ -10,14 +10,20 @@ extends 'DBIx::Class::ResultSet';
 our $VERSION = '0';
 
 sub search_autoqc {
-  my ($self, $values, $size) = @_;
+  my ($self, $query, $size) = @_;
 
   my $how = {'cache' => 1};
+  # Remap the query so that we do not change the input data,
+  # thus allowing the caller to reuse the variable.
+  my %local_values = %{$query};
+  my $values = \%local_values;
   my $rsource = $self->result_source();
   if ($rsource->has_relationship('seq_component_compositions')) {
     foreach my $col_name  (keys %{$values}) {
-      $values->{'seq_component.' . $col_name} = $values->{$col_name};
-      delete $values->{$col_name};
+      if (!$rsource->has_column($col_name)) {
+        $values->{'seq_component.' . $col_name} = $values->{$col_name};
```

Comment:
To be more explicit, why not prepend column names of this table with 'me.'? Th queries will work even if teh two tables have columns with the same name.